src/model/remote-builder/remote-builder.ts

```diff
@@ -0,0 +1,422 @@
+    buildUid: string,
+    buildParameters: BuildParameters,
+    branchName: string | undefined,
+    defaultSecretsArray: { ParameterKey: string; EnvironmentVariable: string; ParameterValue: string }[],
+  ) {
+    core.info('Starting step 1/4 clone and restore cache)');
+    await AWSBuildPlatform.runBuild(
+      buildUid,
+      buildParameters.awsStackName,
+      'alpine/git',
+      [
+        '-c',
+        `apk update;
+          apk add unzip;
+          apk add git-lfs;
+          apk add jq;
+          # Get source repo for project to be built and game-ci repo for utilties
+          git clone https://${buildParameters.githubToken}@github.com/${
+          process.env.GITHUB_REPOSITORY
+        }.git ${buildUid}/${repositoryDirectoryName} -q
+          git clone https://${buildParameters.githubToken}@github.com/game-ci/unity-builder.git ${buildUid}/builder -q
+          git clone https://${buildParameters.githubToken}@github.com/game-ci/steam-deploy.git ${buildUid}/steam -q
+          cd /${efsDirectoryName}/${buildUid}/${repositoryDirectoryName}/
+          git checkout $GITHUB_SHA
+          cd /${efsDirectoryName}/
+          # Look for usable cache
+          if [ ! -d ${cacheDirectoryName} ]; then
+            mkdir ${cacheDirectoryName}
+          fi
+          cd ${cacheDirectoryName}
+          if [ ! -d "${branchName}" ]; then
+            mkdir "${branchName}"
+          fi
+          cd "${branchName}"
+          echo " "
+          echo "Cached Libraries for ${branchName} from previous builds:"
+          ls
+          echo " "
+          ls "/${efsDirectoryName}/${buildUid}/${repositoryDirectoryName}/${buildParameters.projectPath}"
+          libDir="/${efsDirectoryName}/${buildUid}/${repositoryDirectoryName}/${buildParameters.projectPath}/Library"
+          if [ -d "$libDir" ]; then
+            rm -r "$libDir"
+            echo "Setup .gitignore to ignore Library folder and remove it from builds"
+          fi
+          echo 'Checking cache'
+          # Restore cache
+          latest=$(ls -t | head -1)
+          if [ ! -z "$latest" ]; then
+            echo "Library cache exists from build $latest from ${branchName}"
+            echo 'Creating empty Library folder for cache'
+            mkdir $libDir
+            unzip -q $latest -d $libDir
+            # purge cache
+            ${process.env.PURGE_REMOTE_BUILDER_CACHE === '' ? '#' : ''} rm -r $libDir
+          else
+            echo 'Cache does not exist'
+          fi
+          # Print out important directories
+          echo ' '
+          echo 'Repo:'
+          ls /${efsDirectoryName}/${buildUid}/${repositoryDirectoryName}/
+          echo ' '
+          echo 'Project:'
+          ls /${efsDirectoryName}/${buildUid}/${repositoryDirectoryName}/${buildParameters.projectPath}
+          echo ' '
+          echo 'Library:'
+          ls /${efsDirectoryName}/${buildUid}/${repositoryDirectoryName}/${buildParameters.projectPath}/Library/
+          echo ' '
```

Comment:
Is it possible to use empty lines above the comments? or does that terminate execution?

It would make it easier to read I think.

Reply:
empty lines replacing echo would result in no line separation in the logs.

Do you prefer `echo`?